### DISCUSSION TITLE
How to talk the the camera in a secure way ?

### DISCUSSION THREAD

**Discussion by Val9000** [AxisCommunications/discussions#175] [VAPIX]
Hello, I have the following question.

If I for example upload certificates automatically per code, how would I ensure that the transmission of the certificates (keys) are in a secured manner? So that a man in the middle attack is not possible. Because the initial upload of the first certificate can always be fished since by default the camera runs in HTTP... and to enable HTTPS one first has to send a command... and also that can be fished.

I would appreciate any tips on that topic.
Thanks in advance.

**Comment by vivekatoffice** [comment]
Hi @Val9000 ,
You have a valid point here. Please see the upcoming changes in Axis OS 12 here:
[changes-during-axis-os-12](https://help.axis.com/en-us/axis-os#changes-during-axis-os-12)
![image](https://github.com/AxisCommunications/discussions/assets/52032125/d8514e7b-9bf4-42cc-9e56-22b314af3b8c)
